apps/passport-client/new-components/shared/Modals/PodsCollectionBottomModal.tsx

```diff
@@ -121,12 +122,47 @@ export const PodsCollectionList = ({
       });
     }
 
-    return Object.values(result).filter((group) => group.children.length > 0);
-  }, [pcdCollection, onPodClick]);
+    return Object.values(result)
+      .map((group) => {
+        if (!searchQuery) {
+          return group;
+        }
+
+        if (
+          group.title &&
+          group.title.toLowerCase().includes(searchQuery.toLowerCase())
+        ) {
+          return group;
+        }
+
+        return {
+          ...group,
+          children: group.children.filter((pod) =>
+            pod.title.toLowerCase().includes(searchQuery.toLowerCase())
+          )
+        };
+      })
+      .filter((group) => group.children.length > 0);
+  }, [pcdCollection, onPodClick, searchQuery]);
 
-  return <List style={style} list={podsCollectionList} />;
+  return (
+    <>
+      <SearchPodInputContainer>
+        <Input2
+          placeholder="Search for pods..."
+          variant="secondary"
+          onChange={({ target: { value } }) => setSearchQuery(value)}
+        />
```

Comment:
I think we should have this conditionally render based on if we have > N PODs (we can start with N = 10)